pkgs/build-support/build-bazel-package/default.nix

```diff
@@ -76,6 +76,11 @@ in stdenv.mkDerivation (fBuildAttrs // {
       # For example, in Tensorflow-gpu build:
       # platforms -> NIX_BUILD_TOP/tmp/install/35282f5123611afa742331368e9ae529/_embedded_binaries/platforms
       find $bazelOut/external -maxdepth 1 -type l | while read symlink; do
+        # There is a top level symlink that points to the source directory, we need to keep that
+        target="$(readlink "$symlink")"
+        if [[ "$target" == "$NIX_BUILD_TOP/source" ]]; then
```

Comment:
Not all projects have `source` directory. Interesting that it's needed, maybe we need to filter `/tmp` symlinks instead?

Reply:
I think it appears if they have external repositories that refer back to the root repository via `@//path:name` or `@envoy//path:name`.

Is the symlink removal here for symlinks to `NIX_BUILD_TOP/tmp` or to `/tmp`? Because if it's `/tmp` then we can delete all symlinks that point to outside of `NIX_BUILD_TOP`. But even if it points to `NIX_BUILD_TOP/tmp` why is that a problem? As long as it points inside `NIX_BUILD_TOP` rewriting the symlink the build part of the derivation should make it work?

Reply:
The problem is that there is a hash in the path that changes from run to run (see an example above). Maybe we should drop everything to `$NIX_BUILD_TOP/tmp`?

Reply:
Ah yes I see. I would have expected `--output_user_root` to not add the hash. I guess because we control which directory that is we can then drop just the symlinks that point to that directory. Maybe we can rename it while we're at it, `tmp` didn't make it obvious that it was the output user root.